coriolis_openstack_utils/actions.py

```diff
@@ -1090,3 +1091,143 @@ def execute_operations(self):
+            if self.payload['src_network_id'] == (
+                    other_action.payload.get('src_network_id')):
+                return (self.payload.get('dest_tenant_id') ==
+                        other_action.payload.get('dest_tenant_id'))
+        return False
+
+    def print_operations(self):
+        super(NetworkCreationAction, self).print_operations()
+        network_name = self.get_new_network_name()
+        LOG.info(
+            "Create new destination network named '%s'." % network_name)
+
+    def get_source_network_name(self):
+        return networks.get_network(
+            self._source_client, self.payload['src_network_id'])['name']
+
+    def get_new_network_name(self):
+        return self.network_name_format % {
+            "original": self.get_source_network_name()}
+
+    def create_network_body(self, description):
+        src_body = networks.get_body(
+            self._source_client, self.payload['src_network_id'])
+        dest_network_name = self.get_new_network_name()
+        dest_tenant_id = self.payload['dest_tenant_id']
+        body = {'name': dest_network_name,
+                'tenant_id': dest_tenant_id,
+                'project_id': dest_tenant_id,
+                'description': description}
+
+        for k in src_body:
+            if k not in body and k != 'subnets':
+                body[k] = src_body[k]
+
+        return body
+
+    def execute_operations(self):
+        super(NetworkCreationAction, self).print_operations()
+        done = self.check_already_done()
+        dest_network_name = self.get_new_network_name()
+        if done["done"]:
+            LOG.info(
+                "Network named '%s' already exists.",
+                dest_network_name)
+            return done["result"]
+
+        LOG.info("Creating destination Network with name '%s'" %
+                 dest_network_name)
+
+        description = (self.NEW_NETWORK_DESCRIPTION %
+                       self.get_source_network_name())
+
+        body = self.create_network_body(description)
+        # TODO: create subnets from destination
+        dest_network_id = networks.create_network(
+            self._destination_client, body)
+
+        src_subnet_ids = networks.get_network(
+            self._source_client, self.payload['src_network_id'])['subnets']
+
+        src_subnet_names = [
+            subnets.get_subnet(self._source_client, subnet_id)['name']
+            for subnet_id in src_subnet_ids]
+
+        for name in src_subnet_names:
+            subnet_migration_payload = {
+                'source_name': name,
+                'src_network_id': self.payload['src_network_id'],
+                'dest_network_id': dest_network_id}
+            self.subactions.append(
+                SubnetCreationAction(
+                    self._source_client, self._destination_client,
+                    subnet_migration_payload))
+
+        for subaction in self.subactions:
```

Comment:
Why not just call `.execute_operations()` on the subaction in the above `for` loop?